spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala

```diff
@@ -309,8 +309,9 @@ case class TypeWideningPreDowngradeCommand(table: DeltaTableV2)
    * @return Return the number of files rewritten.
    */
   private def rewriteFilesIfNeeded(): Long = {
-    val numFilesToRewrite = TypeWidening.numFilesRequiringRewrite(table.initialSnapshot)
-    if (numFilesToRewrite == 0L) return 0L
+    if (!TypeWideningMetadata.containsTypeWideningMetadata(table.initialSnapshot.schema)) {
```

Comment:
Is it guaranteed that there are no files to rewrite if the metadata is not contained? 
I guess if there is no metadata but there are files the table is not fully readable anyways?